### ISSUE TITLE
A couple more

### ISSUE THREAD

**Issue by TrevorBasinger** [fantasyland/fantasy-sorcery#6]
Hey,

I've got a couple more commonly functions that I didn't see implemented in here. I thought about putting together a PR, but I wanted to get your thoughts first.

Straight from my source:

```
konst   = R.curry (function (a, b) { return a; });
du      = function (M) { return function () { return R.apply (R.pipe, arguments)(M.of ({})); }; };
bind    = function (a) { return R.chain (konst (a)); };
```

I find myself using these a lot. It allows me to write code that looks like this:

```
//  describePrivateIps :: AWS [Maybe String]
    describePrivateIps = du(A)( bind  (A.setRegion ('us-west-1'))
                              , bind  (A.describeInstances ({}))
                              , map   (R.prop ('Reservations'))
                              , map   (map (R.prop ('Instances')))
                              , map   (R.unnest)
                              , map   (map (R.compose (Maybe.fromNullable, R.prop ('PrivateIpAddress'))))),
```

Do you think any of these functions have a place inside of fantasy-sorcery?
If so, I'll put together a PR for you.

**Comment by SimonRichardson** [comment]
@TrevorBasinger thanks!

- `konst` : https://github.com/fantasyland/fantasy-combinators/blob/master/src/constant.js
- `du` : What does this do?
- `bind` : useful, can you make a PR.

**Comment by TrevorBasinger** [comment]
@SimonRichardson du allows you to have a weak version of "do notation". Basically, it takes a Monad that follows the fantasy land spec and allows you to sequence over monads.

The flow is basically this:
1 - Accept monad and unspecified length of functions that accept a monad and return a monad.
2 - Compose monadic functions with a little javascript apply magic.
3 - Create an instance of the monad that contains a unit value.
4 - Pass that monad into the composition.

Basically, I always have `bind` as my first step. It accepts the unit value wrapped in a monad and then discards it. It then continues down the sequence and results in a transformed monadic value.

It's really been helpful and makes things easier to read. My coworkers have also been extremely pleased with how much more readable it makes things.

**Comment by adam8810** [comment]
:+1:

**Comment by davidchambers** [comment]
This project is no longer maintained. See #9 for details.
